pkg/microservice/aslan/core/workflow/service/workflow/workflow_task_v4.go

```diff
@@ -855,6 +866,257 @@ func ListWorkflowTaskV4(workflowName string, pageNum, pageSize int64, logger *za
+								if result != nil && result.FunctionTestSuite != nil {
+									duration := 0.0
+									for _, testCase := range result.FunctionTestSuite.TestCases {
+										duration += testCase.Time
+									}
+									testModule := &commonmodels.WorkflowTestModule{
+										RunningJobName: runningJob.Name,
+										Type:           "function",
+										TestName:       result.FunctionTestSuite.Name,
+										TestCaseNum:    result.FunctionTestSuite.Tests,
+										SuccessCaseNum: result.FunctionTestSuite.Successes,
+									}
+									logger.Infof("----->test_name:%s", testModule.TestName)
+									if testModule.TestName == "" {
+										keys := strings.Split(runningJob.Key, ".")
+										testModule.TestName = keys[len(keys)-1]
+										logger.Infof("----->>test_name:%s", testModule.TestName)
+									}
+									if result.FunctionTestSuite.Time == 0 {
+										result.FunctionTestSuite.Time = math.Round(duration*1000) / 1000
+									}
+									testModule.TestTime = result.FunctionTestSuite.Time
+									testModules = append(testModules, testModule)
+								}
+							}
+						}
+					}
+					jobPreview.TestModules = testModules
+				case config.JobZadigDistributeImage:
+					distribute := new(commonmodels.ZadigDistributeImageJobSpec)
+					if err := commonmodels.IToi(job.Spec, distribute); err != nil {
+						return nil, 0, err
+					}
+					serviceModules := make([]*commonmodels.WorkflowServiceModule, 0)
+					for _, target := range distribute.Targets {
+						sm := &commonmodels.WorkflowServiceModule{
+							ServiceName:   target.ServiceName,
+							ServiceModule: target.ServiceModule,
+						}
+						serviceModules = append(serviceModules, sm)
+					}
+				}
+				stagePreview.Jobs = append(stagePreview.Jobs, jobPreview)
+			}
+			if len(stagePreview.Jobs) > 0 {
+				stagePreviews = append(stagePreviews, stagePreview)
+			}
+		}
+
+		for _, stage := range task.Stages {
+			for _, stagePreview := range stagePreviews {
+				if stagePreview.Name == stage.Name {
+					stagePreview.Status = stage.Status
+					stagePreview.StartTime = stage.StartTime
+					stagePreview.EndTime = stage.EndTime
+					stagePreview.Approval = stage.Approval
+					stagePreview.Parallel = stage.Parallel
+					stagePreview.Error = stage.Error
+					break
+				}
+			}
+		}
+		preview.Stages = stagePreviews
+		taskPreviews = append(taskPreviews, preview)
+	}
+	cleanWorkflowV4TasksPreviews(taskPreviews)
+	return taskPreviews, total, nil
+}
+
+// clean extra message for list workflow
+func cleanWorkflowV4TasksPreviews(workflows []*commonmodels.WorkflowTaskPreview) {
+	const StatusNotRun = ""
+	for _, workflow := range workflows {
+		var stageList []*commonmodels.StagePreview
+		workflow.WorkflowArgs = nil
+		for _, stage := range workflow.Stages {
+			if stage.Approval != nil && stage.Approval.Enabled {
+				approvalStage := &commonmodels.StagePreview{
+					Name: map[config.ApprovalType]string{
```

Comment:
this would cause panic if `stage.Approval.Type` not exists in the temp static map, i don't recommend this style unless your key is static too.